File: app/src/main/java/linusfessler/alarmtiles/sleeptimer/events/SleepTimerEvent.java
package linusfessler.alarmtiles.sleeptimer.events;

import com.spotify.mobius.Next;

import linusfessler.alarmtiles.sleeptimer.SleepTimer;
import linusfessler.alarmtiles.sleeptimer.SleepTimerEffect;

public interface SleepTimerEvent {

    Next<SleepTimer, SleepTimerEffect> update(SleepTimer sleepTimer);
}

File: app/src/main/java/linusfessler/alarmtiles/tiles/sleeptimer/SleepTimerEvent.kt
package linusfessler.alarmtiles.tiles.sleeptimer

import linusfessler.alarmtiles.shared.data.Time

interface SleepTimerEvent {
    data class VolumeChanged(val volume: Int) : SleepTimerEvent

    class Initialize : SleepTimerEvent

    data class Initialized(val sleepTimer: SleepTimer) : SleepTimerEvent

    data class Start(val duration: Time) : SleepTimerEvent

    data class StartWith(val startTimestamp: Long, val duration: Time) : SleepTimerEvent

    data class SetDecreasingVolume(val decreasingVolume: Boolean) : SleepTimerEvent

    class Cancel : SleepTimerEvent

    class Finish : SleepTimerEvent

    class FinishWith(val sleepTimer: SleepTimer) : SleepTimerEvent
}